builder/amazon/common/state.go

```diff
@@ -38,12 +40,45 @@ type StateChangeConf struct {
 // Following are wrapper functions that use Packer's environment-variables to
 // determine retry logic, then call the AWS SDK's built-in waiters.
 
-func WaitUntilAMIAvailable(ctx aws.Context, conn ec2iface.EC2API, imageId string) error {
+// Polling configuration for the AWS waiter. Configures the waiter for resources creation or actions like attaching
+// volumes or importing image.
+// Usage example:
+//
+// In JSON:
+// ```json
+// "aws_polling" : {
+// 	 "delay_seconds": 30,
+// 	 "max_attempts": 50
+// }
+// ```
+//
+// In HCL2:
+// ```hcl
+// aws_polling {
+//	 delay_seconds = 30
+//	 max_attempts = 50
+// }
+// ```
+//
+type AWSPollingConfig struct {
+	// Specifies the maximum number of attempts the waiter will check for resource state.
+	// This value can also be set via the AWS_MAX_ATTEMPTS.
+	// If both option and environment variable are set, the max_attepmts will be considered over the AWS_MAX_ATTEMPTS.
+	// If none is set, defaults to AWS waiter default which is 40.
```

Comment:
```suggestion
	// If none is set, defaults to AWS waiter default which is 40 attempts.
```

Reply:
is not me if there aren't any switched letters 😆

Reply:
ops, I wanted to comment the other suggestion :D

Reply:
yes, no worries and no worries :D